src/cloud_provider/digitalocean/router.rs

```diff
@@ -0,0 +1,494 @@
+                self.engine_error_scope(),
+                self.context.execution_id(),
+                crate::template::copy_non_template_files(
+                    format!(
+                        "{}/digitalocean/charts/nginx-ingress",
+                        self.context().lib_root_dir()
+                    ),
+                    into_dir.as_str(),
+                ),
+            )?;
+
+            // do exec helm upgrade and return the last deployment status
+            let helm_history_row = cast_simple_error_to_engine_error(
+                self.engine_error_scope(),
+                self.context.execution_id(),
+                crate::cmd::helm::helm_exec_with_upgrade_history_with_override(
+                    kubernetes_config_file_path.as_str(),
+                    environment.namespace(),
+                    format!("custom-{}", helm_release_name).as_str(),
+                    into_dir.as_str(),
+                    format!("{}/nginx-ingress.yaml", into_dir.as_str()).as_str(),
+                    do_credentials_envs.clone(),
+                ),
+            )?;
+
+            // check deployment status
+            if helm_history_row.is_none() || !helm_history_row.unwrap().is_successfully_deployed() {
+                return Err(self.engine_error(
+                    EngineErrorCause::Internal,
+                    "Router has failed to be deployed".into(),
+                ));
+            }
+
+            // waiting for the nlb, it should be deploy to get fqdn
+            let external_ingress_hostname_custom_result = retry::retry(
+                Fibonacci::from_millis(3000).take(10),
+                || {
+                    let external_ingress_hostname_custom =
+                        crate::cmd::kubectl::kubectl_exec_get_external_ingress_hostname(
+                            kubernetes_config_file_path.as_str(),
+                            environment.namespace(),
+                            format!(
+                                "app=nginx-ingress,component=controller,release=custom-{}",
+                                helm_release_name
+                            )
+                            .as_str(),
+                            do_credentials_envs.clone(),
+                        );
+
+                    match external_ingress_hostname_custom {
+                        Ok(external_ingress_hostname_custom) => {
+                            OperationResult::Ok(external_ingress_hostname_custom)
+                        }
+                        Err(err) => {
+                            error!(
+                                "Waiting Digital Ocean LoadBalancer endpoint to be available to be able to configure TLS"
+                            );
+                            OperationResult::Retry(err)
+                        }
+                    }
+                },
+            );
+
+            match external_ingress_hostname_custom_result {
```

Comment:
is nlb the name of DO for LBs? If no please update to keep the correct naming convention

Reply:
done